app/forms.py

```diff
@@ -0,0 +1,53 @@
+from flask import Markup, render_template
+from flask_wtf import FlaskForm as Form
+from notifications_utils.formatters import strip_all_whitespace
+from notifications_utils.recipients import (
+    InvalidEmailError,
+    validate_email_address,
+)
+from wtforms import StringField, ValidationError
+from wtforms.validators import DataRequired
+
+
+class ValidEmail:
+    message = 'Not a valid email address'
+
+    def __call__(self, form, field):
+        if not field.data:
+            return
+
+        try:
+            validate_email_address(field.data)
+        except InvalidEmailError:
+            raise ValidationError(self.message)
+
+
+class EmailAddressField(StringField):
+    def widget(self, field, **kwargs):
+        if field.errors:
+            error_message = {"text": field.errors[0]}
+        else:
+            error_message = None
+
+        params = {
+            "id": self.id,
+            "name": self.name,
+            "type": "email",
+            "errorMessage": error_message,
+            "label": {
+                "text": "Email address",
+                "for": self.id,
+            },
+            "spellcheck": False,
+            "autocomplete": "email",
+            "value": field.data
+        }
+        return Markup(render_template("components/govuk_input.html", params=params))
+
+
+class EmailAddressForm(Form):
+    email_address = EmailAddressField(
+        'Email address',
+        validators=[DataRequired('Enter your email address'), ValidEmail()],
```

Comment:
Hmm ... I wonder if 'your' is the right word here, because theoretically it might not be _their_ email address. It could be a group inbox or the email of someone who's shared the link.

We should maybe put 'Enter _an_ email address'? We can probably leave this as-is but might be good to flag with @karlchillmaid when he's back, or @saimaghafoor for thoughts

Reply:
Could it be 'enter your email address' and then some microcopy around which email address it is? So if it's the one used to register for something or where a link was sent to (or whatever action), this could be mentioned in the microcopy. Example 'Enter your email address' Microcopy 'This must be the email address used when registering/where the document link was sent [action]'.

Reply:
Well, we do already have some copy on the page for that, so I think that may be a bit repetitive?

Here's the current page in the no-error state:
![image](https://user-images.githubusercontent.com/2920760/188421091-b8122e4d-8c01-464b-9bce-2f1f810dcd8c.png)

Reply:
Got it.and completely agree. Sorry the journey didn't quite click before this for me. So I can see where the error would appear. From the screenshots above, do the 2 error messages appear for different reasons?

Reply:
Yes, 'Enter your/an email address' is shown if the user submits the form having put nothing into the field.

'Not a valid email address' appears if they enter _some_ text but it's not a valid email address.

Reply:
Fab thanks. I think because we already state "Confirm your email address" in the heading, it should be OK to use it in the error message too. Unless we go and change both to confirm an email address. But I think I prefer the 'your' as it's more specific than 'an' which seems a random email address will suffice! Hope this makes sense.

Reply:
Ok, I'm still concerned that saying 'your' will be misleading for some people because it's not necessarily _their_ email address.

But we can leave it like that if you're happy.

Reply:
If the error message was appearing on a separate page then I think we could look at it. But because the microcopy states which email address this needs to be, I'm happy to continue with 'your email address'. We can revisit this if it causes issues :)